src/pp-annotator/form/PrzypisEditor.tsx

```diff
@@ -85,68 +87,57 @@ export default class PrzypisEditor extends Widget {
   }
 
   /**
-   * Override parent attach function to render React form
+   * Loads the annotation and displays the edit window
+   * Returns an unresolved Promise that will be resolved/rejected when the save/cancel button is clicked.
    */
-  public attach() {
-    // Call parent function (renders PrzypisEditor.template)
-    super.attach();
-    this.updateForm({});
-  }
-
-  /**
-   * Returns an unresolved Promise that will be resolved when the save/cancel button is clicked.
-   * If load function is waited upon, it will finish only when the save/cancel button is clicked.
-   */
-  public load(annotation: annotator.IAnnotation, position: util.IPosition) {
+  public load(annotation: annotator.IAnnotation,
+              position: util.IPosition,
+              saveAction: (annotation: IAnnotation) => any) {
     this.annotation = annotation;
-    this.updateForm(annotation.fields || {});
+    this.saveAction = saveAction.bind(this);
+    this.updateForm(annotation);
+    this.show(position);
 
     return new Promise<IAnnotation>((resolve, reject) => {
       this.promiseResultContainer = {
```

Comment:
Saving Promise resolution methods like this for later resolution is a serious antipattern — it makes the flow extremely unpredictable. Since we're reworking the flow here, this might be a good place to resolve this issue.